mlflow/keras.py

```diff
@@ -144,24 +158,42 @@ def log_model(keras_model, artifact_path, conda_env=None, **kwargs):
     >>>   mlflow.keras.log_model(keras_model, "models")
     """
     Model.log(artifact_path=artifact_path, flavor=mlflow.keras,
-              keras_model=keras_model, conda_env=conda_env, **kwargs)
+              keras_model=keras_model, conda_env=conda_env, custom_objects=custom_objects, **kwargs)
+
 
+def _save_custom_objects(path, custom_objects):
+    import cloudpickle
+    custom_objects_subpath = _CUSTOM_OBJECTS_SAVE_PATH
+    custom_objects_path = os.path.join(path, custom_objects_subpath)
+    with open(custom_objects_path, "wb") as out_f:
+        cloudpickle.dump(custom_objects, out_f)
+    return custom_objects_subpath
 
-def _load_model(model_file, **kwargs):
+
+def _load_model(model_path, **kwargs):
     import keras
     import keras.models
     import h5py
-
+    custom_objects = kwargs.get("custom_objects", {})
+    custom_objects_path = None
+    if os.path.isdir(model_path):
+        if os.path.isfile(os.path.join(model_path, _CUSTOM_OBJECTS_SAVE_PATH)):
+            custom_objects_path = os.path.join(model_path, _CUSTOM_OBJECTS_SAVE_PATH)
+        model_path = os.path.join(model_path, _MODEL_SAVE_PATH)
+    if custom_objects_path is not None:
+        import cloudpickle
+        with open(custom_objects_path, "rb") as in_f:
+            pickled_custom_objects = cloudpickle.load(in_f)
+            custom_objects.update(pickled_custom_objects)
```

Comment:
This update ordering here will ignore user-specified `custom_objects` if they are already present in `pickled_custom_objects`. I think we should preferentially select the user-specified `"custom_objects"` dictionary instead by reversing the update order (e.g., `pickled_custom_objects.update(custom_objects)`).

Reply:
Can we also add a test to ensure that user-specified custom objects are preferentially selected?

Reply:
Great idea! Done